test/unit/detectRootFiles.test.js

```diff
@@ -0,0 +1,432 @@
+const expect = require('chai').expect,
+  Converter = require('../../index.js'),
+  fs = require('fs'),
+  path = require('path'),
+  COUNTING_SEPARATED_FOLDER = '../data/separatedFiles/counting',
+  WIKI_20_FOLDER = '../data/separatedFiles/wiki20',
+  MULTIPLE_ROOT = '../data/separatedFiles/multipleRoot';
+
+describe('detectRoot method', function() {
+
+  it('should return one root 1.1 correctly no specific version', async function() {
+    const service = path.join(
+        __dirname,
+        COUNTING_SEPARATED_FOLDER,
+        '/CountingCategoryService.wsdl'
+      ),
+      types = path.join(
+        __dirname,
+        COUNTING_SEPARATED_FOLDER,
+        '/CountingCategoryData.xsd'
+      );
+    let serviceContent = fs.readFileSync(service, 'utf8'),
+      typesContent = fs.readFileSync(types, 'utf8'),
+      input = {
+        type: 'multiFile',
+        data: [
+          {
+            path: '/CountingCategoryService.wsdl',
+            content: serviceContent
+          },
+          {
+            path: '/CountingCategoryData.xsd',
+            content: typesContent
+          }
+        ]
+      },
+      result = await Converter.detectRootFiles(input);
+    expect(result.result).to.be.true;
+    expect(result.output.data[0].path).to.equal('/CountingCategoryService.wsdl');
+    expect(result.output.specification.version).to.equal('1.1');
+    expect(result.output.type).to.be.equal('rootFiles');
+  });
+
+  it('should return one root 1.1 correctly', async function() {
+    const service = path.join(
+        __dirname,
+        COUNTING_SEPARATED_FOLDER,
+        '/CountingCategoryService.wsdl'
+      ),
+      types = path.join(
+        __dirname,
+        COUNTING_SEPARATED_FOLDER,
+        '/CountingCategoryData.xsd'
+      );
+    let serviceContent = fs.readFileSync(service, 'utf8'),
+      typesContent = fs.readFileSync(types, 'utf8'),
+      input = {
+        type: 'multiFile',
+        specificationVersion: '1.1',
+        data: [
+          {
+            path: '/CountingCategoryService.wsdl',
+            content: serviceContent
+          },
+          {
+            path: '/CountingCategoryData.xsd',
+            content: typesContent
+          }
+        ]
+      },
+      result = await Converter.detectRootFiles(input);
+    expect(result.result).to.be.true;
+    expect(result.output.data[0].path).to.equal('/CountingCategoryService.wsdl');
+    expect(result.output.specification.version).to.equal('1.1');
+    expect(result.output.type).to.be.equal('rootFiles');
+  });
+
+  it('should return no root when specific version is not present', async function() {
```

Comment:
should be:

should return no root when a specific version is 2.0 and no root of that version is present